lib/esp-scanning-lib.robot

```diff
@@ -0,0 +1,126 @@
+    END
+    IF    "${type}"=="manager"
+        SSHLibrary.Execute Command    mkdir ${folder_name}
+        SSHLibrary.Execute Command    wget ${SHIMX64_URL} -O ${folder_name}/elilo.efi
+    END
+    IF    "${type}"=="linux"
+        SSHLibrary.Execute Command    mkdir ${folder_name}
+        SSHLibrary.Execute Command    wget ${SHIMX64_URL} -O ${folder_name}/shimx64.efi
+    END
+    SSHLibrary.Execute Command    wget ${GRUBX64_URL} -O ${folder_name}/grubx64.efi
+
+    # since switch to root user doesn't really wanna work
+    SSHLibrary.Execute Command    cp -r ~/${folder_name}/ /boot/efi/EFI/
+    ...    sudo=True
+    ...    sudo_password=ubuntu
+
+    SSHLibrary.Execute Command    sync
+    ...    sudo=True
+    ...    sudo_password=ubuntu
+
+    SSHLibrary.Execute Command    rm ${folder_name} -r
+
+Check Boot Menu For All Supported Systems
+    [Documentation]    This keyword scans the boot menu in search for all
+    ...    systems supported by the ESP scan feature. To be used after
+    ...    populating ESP with all the required files.
+    ...    Available modes:
+    ...    normal
+    ...    empty - keyword fails if the list is not empty
+    [Arguments]    ${mode}=normal
+
+    ${boot_list}=    Get Boot Menu Construction
+    Press Key N Times    1    ${ARROW_UP}
+    IF    "${mode}"!="empty"
+        ${boot_list}=    Get Setup Menu Construction    Debian
+    END
+    IF    "${mode}"=="empty"
+        Only N Occurrences    ${boot_list}    0    *Suse*
+        Only N Occurrences    ${boot_list}    0    *RedHat*
+    ELSE
+        Only N Occurrences    ${boot_list}    2    *Suse*
+        Only N Occurrences    ${boot_list}    2    *RedHat*
+    END
+    @{systems}=    Create List    Fedora    OpenSuse    Windows    CentOS    QubesOS
+    FOR    ${system}    IN    @{systems}
+        IF    "${mode}"=="empty"
+            Should Not Contain Match    ${boot_list}    *${system}*
+        END
+        IF    "${mode}"=="normal"
+            Should Contain Match    ${boot_list}    *${system}*
+        END
+        IF    "${mode}"=="double_entry_check"
+            Only N Occurrences    ${boot_list}    1    *${system}*
+        END
+    END
+
+Only N Occurrences
+    [Documentation]    This keyword ensures that there's Only N Occurrences
+    ...    of a given entry in a list
+    [Arguments]    ${list}    ${n}    ${entry}
+    ${x}=    Get Match Count    ${list}    ${entry}
+    Should Be Equal As Integers    ${x}    ${n}
+
+Remove All Supported Systems From Efi
+    Remove System From Efi    Fedora
+    Remove System From Efi    Suse
+    Remove System From Efi    opensuse
+    Remove System From Efi    Microsoft
+    Remove System From Efi    Redhat
+    Remove System From Efi    Centos
+    Remove System From Efi    qubes
+    Remove System From Efi    DTS
+    Remove System From Efi    debian
+
+Remove System From Efi
+    [Arguments]    ${folder_name}
+    SSHLibrary.Execute Command    rm -r /boot/efi/EFI/${folder_name}
```

Comment:
This is executed on the DUT?
We should not use SSHLibrary directly for this. See: https://github.com/Dasharo/open-source-firmware-validation#guidelines

Reply:
All instances replaced with Execute Command In Terminal